### ISSUE TITLE
CV06: require_final_semicolon fails for all procedures in 4.0.4

### ISSUE THREAD

**Issue by sebastien-irco** [sqlfluff/sqlfluff#7533]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

After upgrading from version 3.5.0 to 4.0.4, CV06 suddenly fails for all procedures (but no other objects, e.g. functions, views). It just appears as if the semicolons at the end of procedures are no longer recognised.

Minimal example:

SQLFluff: 4.0.4

Config:

```
[sqlfluff]
dialect = bigquery
rules = convention.terminator

[sqlfluff:rules:convention.terminator]
require_final_semicolon = True
```

Procedure (fails):

```sql
CREATE OR REPLACE PROCEDURE test_proc ()
BEGIN
  SELECT 1;
END;
```

```bash
sqlfluff lint test_proc.sql
== [test_proc.sql] FAIL
L:   1 | P:   1 | CV06 | Statements must end with a semi-colon.
                       | [convention.terminator]
All Finished 📜 🎉!
```

Table function (passes):

```sql
CREATE OR REPLACE TABLE FUNCTION test_func ()
AS
SELECT 1;
```

```bash
sqlfluff lint test_func.sql
All Finished 📜 🎉!
```

Also, running `sqlfluff fix` shows the behavior that I already found in https://github.com/sqlfluff/sqlfluff/issues/4924

```bash
sqlfluff fix test_proc.sql
==== finding fixable violations ====
== [test_proc.sql] FAIL
L:   1 | P:   1 | CV06 | Statements must end with a semi-colon.
                       | [convention.terminator]
== [test_proc.sql] FIXED
1 fixable linting violations found

sqlfluff fix test_proc.sql
==== finding fixable violations ====
== [test_proc.sql] FAIL
L:   1 | P:   1 | CV06 | Statements must end with a semi-colon.
                       | [convention.terminator]
== [test_proc.sql] FIXED
1 fixable linting violations found
```

Outcome (one semicolon added for every run):

```sql
CREATE OR REPLACE PROCEDURE test_proc ()
BEGIN
  SELECT 1;
END;;;
```

### Expected Behaviour

The bug introduces between versions 3.5.0 and 4.0.4 should be fixed so that `sqlfluff lint` no longer fails and `sqlfluff fix` no longer adds semicolons, when there is already one.

### Observed Behaviour

See above

### How to reproduce

See above

### Dialect

BigQuery

### Version

Python 3.12.2
sqlfluff, version 4.0.4

### Configuration

see above

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
#4924 was alternating: once added then removed the terminator :) This bug keeps adding...

My hunch is that the reason is that the PROCEDURE statement in BigQuery grammar has a type of `create_procedure_statement`
